### PR TITLE
refactor: Make flobject consistent wrt server.

### PR DESCRIPTION
**Pull request by prmukherj**
Please note:
Along with this there should be a couple of minor style fixes in the server side to have the 2 files completely in sync.

There has been some changes in the rename method to support backwards compatibility. 
Tests have been added to check the earlier and latest behavior of 'rename' method. Server-side flobject.py will be updated accordingly.

### CODE REVIEW

**Review by mkundu1 — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

**Review by gyeole — APPROVED** [review]

File: src/ansys/fluent/core/solver/flobject.py [hunk 1]
```diff
@@ -1292,9 +1309,6 @@ def _process_cls_names(info_dict, names, write_doc=False):
             cls.child_object_type = get_cls(
                 "child-object-type", object_type, cls, version=version
             )
-            cls.child_object_type.rename = lambda self, name: self._parent.rename(
-                name, self._name
-            )
```
Comment (prmukherj) [inline src/ansys/fluent/core/solver/flobject.py:1297]: This was added in server side to maintain backwards compatibility. However, it is not required as the backend settings-api framework has not changed and thus it handles the earlier cases as well.
The server-side code will also be updated accordingly.

Thank you @gyeole and @V-Dandekar for your inputs.

File: tests/test_flobject.py [hunk 2]
```diff
@@ -707,6 +701,12 @@ def test_accessor_methods_on_settings_object(load_static_mixer_settings_only):
     else:
         assert not mesh.name.is_read_only()
 
+    assert solver.results.graphics.mesh.get_object_names() == ["mesh-1"]
+
+    solver.results.graphics.mesh.rename("mesh_new", "mesh-1")
+
+    assert solver.results.graphics.mesh.get_object_names() == ["mesh_new"]
+
```
Comment (prmukherj) [inline tests/test_flobject.py:709]: This test checks the working of rename. Both the new behavior and the backwards compatibility will be checked via this.

**Review by V-Dandekar — APPROVED** [review]

**Comment by seanpearsonuk** [comment]
@prmukherj please also bring the Fluent file imediately up to date. There seem to be quite a lot of diffs. Thanks.

**Comment by prmukherj** [comment]
@seanpearsonuk, I'll update the fluent side along with this only.

**Review by seanpearsonuk — APPROVED** [review]

**Pull request merged**